woocommerce/woocommerce-gateway-stripe
# Map variations of German to the standard German locale for Stripe customers

Fixes #2114

## Changes proposed in this Pull Request:

Map the following locales to standard German (de-DE):
- de_DE_formal (Formal German)
- de_CH (Swiss German)
- de_CH_informal (Informal Swiss German)

## Testing instructions

1. Confirm your site language isn't German
2. Log in as a shopper and make a purchase
3. Locate the Stripe customer for this shopper in the Stripe dashboard `https://dashboard.stripe.com/test/customers/<customer ID>`. 
  Or: Go to your Stripe dashboard -> [Payments](https://dashboard.stripe.com/test/payments) -> Open the last payment -> Click on the customer name under "Customer".
4.  Under "Details" at the left, confirm that the value under "Language" isn't German
5. As an admin, change the site language to either of these:  `Deutsch (Schweiz)` - (de_CH), `Deutsch (Schweiz, Du)` - (de_CH_informal), `Deutsch (Sie)` - (de_DE_formal)
6. Make another purchase with the same shopper
7. Go to the Stripe customer page
8. Confirm that the value under "Language" is German
<img width="295" alt="image" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/41606954/e70b38cf-f732-49f4-87c7-c86cd024b137">


I included Swiss German because[ it looks ](https://studyinginswitzerland.com/swiss-german-vs-german-differences/#:~:text=grasp%20of%20it.-,Common%20vs.%20Official,-In%20exploring%20the)like it can be switched with standard German, but glad to remove it if anyone has an objection.

The already opened PR - https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2357 - looks good to me, but I'm opening a new one as I'm not sure how responsive the author would be as the PR is over a year old.

---


**Post merge**
